tests/h/services/search_index/_queue_test.py

```diff
@@ -0,0 +1,364 @@
+                "Syncing 10 annotations that are different in Elasticsearch",
+            )
+        ]
+        batch_indexer.index.assert_called_once_with(
+            Any.list.containing(annotation_ids[:limit]).only()
+        )
+
+    def test_if_there_are_multiple_jobs_with_the_same_annotation_id(
+        self,
+        annotation_ids,
+        batch_indexer,
+        caplog,
+        es,
+        queue,
+    ):
+        queue.add_all(
+            [annotation_ids[0], annotation_ids[0], annotation_ids[0]],
+            tag="test",
+            scheduled_at=five_minutes_ago,
+        )
+
+        queue.sync()
+
+        # It only retrieves the annotation from Elasticsearch once.
+        es.conn.search.assert_called_once_with(
+            body=Any.dict.containing(
+                {
+                    "query": Any.dict.containing(
+                        {
+                            "ids": {
+                                "values": [annotation_ids[0]],
+                            },
+                        },
+                    ),
+                },
+            ),
+        )
+
+        # It only syncs the annotation to Elasticsearch once.
+        assert caplog.record_tuples == [
+            (
+                Any(),
+                Any(),
+                "Syncing 1 annotations that are missing from Elasticsearch",
+            )
+        ]
+        batch_indexer.index.assert_called_once_with(
+            Any.list.containing([annotation_ids[0]]).only()
+        )
+
+    def test_deleting_multiple_jobs_with_the_same_annotation_id(
+        self,
+        annotations,
+        batch_indexer,
+        caplog,
+        db_session,
+        es,
+        queue,
+    ):
+        queue.add_all(
+            [annotations[0].id, annotations[0].id, annotations[0].id],
+            tag="test",
+            scheduled_at=five_minutes_ago,
+        )
+        es.conn.search.return_value = {
+            "hits": {
+                "hits": [
+                    {
+                        "_id": annotations[0].id,
+                        "_source": {
+                            "updated": (annotations[0].updated).isoformat(),
+                        },
+                    }
+                ],
+            },
+        }
+
+        queue.sync()
+
+        assert caplog.record_tuples == [
+            (
+                Any(),
+                Any(),
+                "Deleting 3 successfully completed jobs from the queue",
+            )
+        ]
+        assert db_session.query(Job).all() == []
+        batch_indexer.index.assert_not_called()
+
+
+one_week_from_now = datetime.datetime.utcnow() + datetime.timedelta(weeks=1)
+
+
+five_minutes_ago = datetime.datetime.utcnow() - datetime.timedelta(minutes=5)
+
+
+limit = 10
+
+
+@pytest.fixture
+def annotations(factories):
+    return factories.Annotation.create_batch(size=20)
+
+
+@pytest.fixture
+def annotation_ids(annotations):
+    return [annotation.id for annotation in annotations]
```

Comment:
The two annotation fixtures here are actually private to the second test, which means `LIMIT` is too, and could be a class variable.

Reply:
Moved these into the class. `LIMIT` needs to remain outside though because it's used by the `queue` fixture